CI/samples.ci/client/petstore/java/test-manual/jersey2-java8/JSONComposedSchemaTest.java

```diff
@@ -46,23 +38,126 @@ public void testOneOfSchemaWithDiscriminator() throws Exception {
+        // Mammal is oneOf whale, zebra and pig, and pig is itself one of BasquePig, DanishPig.
         str = "{ \"className\": \"BasquePig\" }";
         o = json.getContext(null).readValue(str, Mammal.class);
         assertTrue(o.getActualInstance() instanceof BasquePig);
+    }
+
+    @Test
+    public void testOneOfNullable() throws Exception {
+        String str = "null";
+        // 'null' is a valid value for NullableShape because it is nullable.
+        AbstractOpenApiSchema o = json.getContext(null).readValue(str, NullableShape.class);
+        assertNull(o);
+
+        // 'null' is a valid value for ShapeOrNull because it is a oneOf with one of the
+        // children being the null type.
+        o = json.getContext(null).readValue(str, ShapeOrNull.class);
+        assertNull(o);
+
+        // 'null' is not a valid value for the Shape model because it is not nullable.
+        // An exception should be raised.
+        Exception exception = assertThrows(JsonMappingException.class, () -> {
+            json.getContext(null).readValue(str, Shape.class);
+        });
+        assertEquals(exception.getMessage(), "Shape cannot be null");
+    }
+
+    /**
+     * Test payload with more than one discriminator.
+     */
+    @Test
+    public void testOneOfMultipleDiscriminators() throws Exception {
+        // 'shapeType' is a discriminator for the 'Shape' model and
+        // 'triangleType' is a discriminator forr the 'Triangle' model.
+        String str = "{ \"shapeType\": \"Triangle\", \"triangleType\": \"EquilateralTriangle\" }";
+
+        // We should be able to deserialize a equilateral triangle into a EquilateralTriangle class.
+        EquilateralTriangle t = json.getContext(null).readValue(str, EquilateralTriangle.class);
+        assertNotNull(t);
+
+        // We should be able to deserialize a equilateral triangle into a triangle.
+        AbstractOpenApiSchema o = json.getContext(null).readValue(str, Triangle.class);
+        assertNotNull(o);
+        assertTrue(o.getActualInstance() instanceof EquilateralTriangle);
+
+        // We should be able to deserialize a equilateral triangle into a shape.
+        o = json.getContext(null).readValue(str, Shape.class);
+        // The container is a shape, and the actual instance should be a EquilateralTriangle.        
+        assertTrue(o instanceof Shape);
+        assertTrue(o.getActualInstance() instanceof EquilateralTriangle);
+
+        // It is not valid to deserialize a equilateral triangle into a quadrilateral.
+        Exception exception = assertThrows(JsonMappingException.class, () -> {
+            json.getContext(null).readValue(str, Quadrilateral.class);
+        });
+        assertTrue(exception.getMessage().contains("Failed deserialization for Quadrilateral: 0 classes match result"));
+    }
+
+    @Test
+    public void testOneOfNestedComposedSchema() throws Exception {
+        /*
```

Comment:
@wing328 , I have temporarily commented out this code. If I enable it, then deserialization fails. It looks like there is a problem when a compose schema (Drawing) has the `additionalProperties` keyword, in which case the generated Drawing.class extends from HashMap<String, X>